### PR TITLE
Add Digital Tyhoon Dataset and adopt framework to regression

### PR DESCRIPTION
**Pull request by nilsleh**
This PR adds a wrapper for the Digital Typhoon Dataset for Image -> single value regression. In line with that, the PR also adapts the foundation models to include a regression version, which inherts just from classification and overwrites loss function and metric.

### CODE REVIEW

File: src/datasets/digital_typhoon_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import torch
+import kornia.augmentation as K
+from torchgeo.datamodules import DigitalTyphoonDataModule
+
+
+class RegDataAugmentation(torch.nn.Module):
+    def __init__(self, split, size):
+        super().__init__()
+
+        # data already comes normalized between 0 and 1
```
Comment (nilsleh) [inline src/datasets/digital_typhoon_wrapper.py:13]: Is this okay for all models?

File: src/foundation_models/base.py [hunk 2]
```diff
@@ -4,17 +4,25 @@
...
+    'relu': nn.ReLU,
+    'elu': nn.ELU,
+}
+
+
 class LinearHead(nn.Module):
     """Linear Classification Head that is attached on top of feature extractors."""
 
-    def __init__(self, in_features: int, num_classes: int):
+    def __init__(self, in_features: int, num_classes: int, activation: str | None = None) -> None:
```
Comment (nilsleh) [inline src/foundation_models/base.py:16]: Is this okay @ando-shah ?

File: src/util/misc.py [hunk 3]
```diff
@@ -65,6 +66,12 @@ def cls_metric(dataset_config, output, target):
         acc1, acc5 = timm_accuracy(output, target, topk=(1, 5))
     return acc1, acc5
 
+def reg_metric(dataset_config, output, target):
```
Comment (nilsleh) [inline src/util/misc.py:69]: Any other metrics you think should be added?

File: src/foundation_models/base.py (hunk 2 above)
Comment (ando-shah) [reply src/foundation_models/base.py:16]: perfect!

File: src/datasets/digital_typhoon_wrapper.py (hunk 1 above)
Comment (ando-shah) [reply src/datasets/digital_typhoon_wrapper.py:13]: Does it need the K.Normalize if its already normed?

Comment (nilsleh) [reply src/datasets/digital_typhoon_wrapper.py:13]: No, not really, but I kept it there for now to see whether we should compute dataset statistics and update. But if [0, 1] range is okay, then we can also drop it

**Comment by nilsleh** [comment]
With the given dataset config I can now launch a run.

File: src/datasets/digital_typhoon_wrapper.py (hunk 1 above)
Comment (ando-shah) [reply src/datasets/digital_typhoon_wrapper.py:13]: Yeah I see your point, esp since most models will be used to zero norm. Lets keep it as-is and I can update with a dataset mean and stdev

File: src/util/misc.py (hunk 3 above)
Comment (ando-shah) [reply src/util/misc.py:69]: Top-5 isnt really useful. The others Id like to have are:
mAP (micro, macro)
F1 (micro, macro)

Comment (nilsleh) [reply src/util/misc.py:69]: I just added regression metrics in this PR, so MSE and MAE, but feel free to add the others, easiest is probably torchmetrics functional calls I would guess
